bitcoin/src/consensus/encode.rs

```diff
@@ -136,6 +136,21 @@ pub fn deserialize_partial<T: Decodable>(data: &[u8]) -> Result<(T, usize), Erro
     Ok((rv, consumed))
 }
 
+/// A trait for integer types smaller than or equal to 64-bits.
+///
+/// We have this trait here because [usize] doesn't implement `Into<u64>`.
+pub trait IntoU64 {
```

Comment:
In e9123d1d25bce9be4d1702ae9da6b99bd2e9674e:

If we are going to make this `pub` it should probably live in the crate root (or maybe even in bitcoin-internals).

As @Kixunil suggests we should tailor our compile-gate by e.g. gating the `usize` impl specifically on `usize` fitting into `u64`. Then only code that tries to use this trait on `usize` will fail to compile on 128-bit arches.